server/src/main/java/org/elasticsearch/action/admin/cluster/storedscripts/GetStoredScriptResponse.java

```diff
@@ -21,39 +21,98 @@
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.StatusToXContentObject;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.rest.RestStatus;
 import org.elasticsearch.script.StoredScriptSource;
 
 import java.io.IOException;
+import java.util.Objects;
 
-public class GetStoredScriptResponse extends ActionResponse implements ToXContentObject {
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
 
+public class GetStoredScriptResponse extends ActionResponse implements StatusToXContentObject {
+
+    public static final ParseField _ID_PARSE_FIELD = new ParseField("_id");
+    public static final ParseField FOUND_PARSE_FIELD = new ParseField("found");
+    public static final ParseField SCRIPT = new ParseField("script");
+
+    private static final ConstructingObjectParser<GetStoredScriptResponse, String> PARSER =
+        new ConstructingObjectParser<>("GetStoredScriptResponse",
+            true,
+            (a, c) -> {
+                String id = (String) a[0];
+                boolean found = (Boolean)a[1];
+                StoredScriptSource scriptSource = (StoredScriptSource)a[2];
+                return found ? new GetStoredScriptResponse(id, scriptSource) : new GetStoredScriptResponse(id);
+            });
+
+    static {
+        PARSER.declareField(constructorArg(), (p, c) -> p.text(),
+            _ID_PARSE_FIELD, ObjectParser.ValueType.STRING);
+        PARSER.declareField(constructorArg(), (p, c) -> p.booleanValue(),
+            FOUND_PARSE_FIELD, ObjectParser.ValueType.BOOLEAN);
+        PARSER.declareField(optionalConstructorArg(), (p, c) -> StoredScriptSource.fromXContent(p, true),
+            SCRIPT, ObjectParser.ValueType.OBJECT);
+    }
+
+    private String id;
     private StoredScriptSource source;
 
     GetStoredScriptResponse() {
     }
 
-    GetStoredScriptResponse(StoredScriptSource source) {
+    GetStoredScriptResponse(String id) {
+        this.id = id;
+    }
+
+    GetStoredScriptResponse(String id, StoredScriptSource source) {
+        this(id);
         this.source = source;
     }
 
+    public String getId() {
+        return id;
+    }
+
     /**
      * @return if a stored script and if not found <code>null</code>
      */
     public StoredScriptSource getSource() {
         return source;
     }
 
+    @Override
+    public RestStatus status() {
+        return source != null ? RestStatus.OK : RestStatus.NOT_FOUND;
+    }
+
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        source.toXContent(builder, params);
+        builder.startObject();
 
+        builder.field(_ID_PARSE_FIELD, id);
```

Comment:
we may have a small problem here, when toXContent is called on an object deserialized from a previous version that didn't send the _id .

Reply:
you added id as a member, which is great. It is now being serialized over the wire, but nodes on previous versions don't send it. If you are deserializing from a previous version this one will be null, and what will happen when you call toXcontent against such an object? I am not sure whether it will be NPE or the id will be printed out null, maybe just the latter, can you check?

Reply:
agreed - use `writeOptionalString` instead of `writeString` (to avoid NPE on serialization) - while I don't see anything more how we can handle this case

Reply:
if it prints out null it may be ok, if it gives NPE we need a null check, that's what I meant.

Reply:
it prints out `null` :
```
public XContentBuilder field(String name, String value) throws IOException {
        if (value == null) {
            return nullField(name);
        }
        ensureNameNotNull(name);
        generator.writeStringField(name, value);
        return this;
    }
```